builder/amazon/common/state_test.go

```diff
@@ -0,0 +1,63 @@
+package common
+
+import (
+	"os"
+	"reflect"
+	"testing"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws/request"
+)
+
+func clearEnvVars() {
+	os.Unsetenv("AWS_POLL_DELAY_SECONDS")
+	os.Unsetenv("AWS_MAX_ATTEMPTS")
+	os.Unsetenv("AWS_TIMEOUT_SECONDS")
+}
+
+func testGetWaiterOptions(t *testing.T) {
+	clearEnvVars()
+
+	// no vars are set
+	options := getWaiterOptions()
```

Comment:
Go tests are run concurrently in goroutines, so using env vars to control behavior is generally a bit risky. If two tests that read/set an env var are run concurrently, you may get undefined behavior.

When code under test needs to access the environment, I like to split the code into two parts. The first part creates a config object from the environment, and the second part performs some action based on the values in the config. That way, in tests, you can construct the config object directly. In general I think it's safe to assume that standard library code works correctly, so we don't have to worry too much about testing that we can read from the environment okay